### DISCUSSION TITLE
[Edit] in code editors for Windows

### DISCUSSION THREAD

**Discussion by Korb** [erosman/support#639] [Q&A]
Please tell me which desktop HTML / CSS / JS editors have the “Edit”, "bbCodeWebex", and "bbCodeInsert" browser add-on functionality? Maybe there is some kind of [add-on for Microsoft Visual Studio Code](https://marketplace.visualstudio.com/vscode) that provides the same convenience for creating snippets and applying them to selected text, as is implemented in Edit? I mean the ability to specify quite large blocks of code in the add-on settings, including line breaks. I looked at Bluefish, SeaMonkey Composer, Arachnophilia and some proprietary products like Dreamweaver, Oxygen XML Editor, Codelobster, CoffeeCup HTML Editor and HTML-Kit and didn't find what I was looking for.

**Comment by erosman** [comment]
I use VS Code and the HTML tags are built-in.
You can use snippets to customise.

There are many [Surround](https://marketplace.visualstudio.com/search?term=surround&target=VSCode&category=All%20categories&sortBy=Relevance) type extension to surrounding text with a tag.
There are also a few [bbcode](https://marketplace.visualstudio.com/search?term=bbcode&target=VSCode&category=All%20categories&sortBy=Relevance) extensions.

**Reply by Korb** [reply]
> I use VS Code and the HTML tags are built-in.

Yes, it is, as are dozens of other programming and markup languages. This is a good app, quite functional out of the box. That's what caught my attention.

> You can use snippets to customise.

That's what they exist for.

> There are many [Surround](https://marketplace.visualstudio.com/search?term=surround&target=VSCode&category=All%20categories&sortBy=Relevance) type extension to surrounding text with a tag. There are also a few [bbcode](https://marketplace.visualstudio.com/search?term=bbcode&target=VSCode&category=All%20categories&sortBy=Relevance) extensions.

I just came across [Surround](https://marketplace.visualstudio.com/items?itemName=yatki.vscode-surround) by Mehmet Yatkı, which you write about. But if you compare the mechanism for creating a list of snippets in Edit and in Surround, a significant difference is obvious. It’s one thing to click on the “Add” button, paste a piece of code into the left input field, and paste another piece of code into the right field, then click the Save button and use it. And to change the order of snippets, just drag them to the desired place in the list.

And it’s quite another thing to edit the surround.custom file with pieces of code like the following

```
{
  "surround.custom": {
    // command name must be unique
    "yourCommandName": {
      // label must be unique
      "label": "Your Snippet Label",
      "description": "Your Snippet Description",
      "snippet": "burrito { $TM_SELECTED_TEXT }$0", // <-- snippet goes here.
      "languageIds": ["html", "javascript", "typescript", "markdown"]
    },
    // You can add more ...
  }
}
```

Surround is much more difficult to understand for a layout designer, but not a programmer. However, I will continue to search for an extension that more closely matches my idea of ​​convenience.

So, do I understand correctly that out of the box no HTML code editing application supports a user-friendly interface for managing wrapping snippets?

**Comment by erosman** [comment]
> So, do I understand correctly that out of the box no HTML code editing application supports a user-friendly interface for managing wrapping snippets?

I have not tested all of them.

I used to use UltraEdit for many years and I created my own surround snippets which were very easy to create and insert (from the visible panel, similar to the Edit extension). However, it is not a free program and when I switched to Linux, I moved to VS Code.
